pkgs/by-name/cl/clickhouse/generic.nix

```diff
@@ -0,0 +1,185 @@
+      rm -rf $out/contrib/sysroot/linux-*
+      rm -rf $out/contrib/liburing/man
+
+      # compress to not exceed the 2GB output limit
+      # try to make a deterministic tarball
+      tar -I 'gzip -n' \
+        --sort=name \
+        --mtime=1970-01-01 \
+        --owner=0 --group=0 \
+        --numeric-owner --mode=go=rX,u+rw,a-s \
+        --transform='s@^@source/@S' \
+        -cf temp  -C "$out" .
+      rm -r "$out"
+      mv temp "$out"
+    '';
+  };
+
+  strictDeps = true;
+  nativeBuildInputs =
+    [
+      cmake
+      ninja
+      python3
+      perl
+      llvmPackages_19.lld
+    ]
+    ++ lib.optionals stdenv.hostPlatform.isx86_64 [
+      nasm
+      yasm
+    ]
+    ++ lib.optionals stdenv.hostPlatform.isDarwin [
+      llvmPackages_19.bintools
+      findutils
+      darwin.bootstrap_cmds
+    ]
+    ++ lib.optionals rustSupport [
+      rustc
+      cargo
+      rustPlatform.cargoSetupHook
+    ];
+
+  buildInputs = lib.optionals stdenv.hostPlatform.isDarwin [ libiconv ];
+
+  dontCargoSetupPostUnpack = true;
+
+  postPatch =
+    ''
+      patchShebangs src/
+      patchShebangs utils/
+
+      sed -i 's|/usr/bin/env perl|"${lib.getExe perl}"|' contrib/openssl-cmake/CMakeLists.txt
+
+      substituteInPlace src/Storages/System/StorageSystemLicenses.sh \
+        --replace-fail '$(git rev-parse --show-toplevel)' "$NIX_BUILD_TOP/$sourceRoot"
+      substituteInPlace utils/list-licenses/list-licenses.sh \
+        --replace-fail '$(git rev-parse --show-toplevel)' "$NIX_BUILD_TOP/$sourceRoot"
+    ''
+    + lib.optionalString stdenv.hostPlatform.isDarwin ''
+      sed -i 's|gfind|find|' cmake/tools.cmake
+      sed -i 's|ggrep|grep|' cmake/tools.cmake
+
+      # Make sure Darwin invokes lld.ld64 not lld.
+      substituteInPlace cmake/tools.cmake \
+        --replace '--ld-path=''${LLD_PATH}' '-fuse-ld=lld'
+    ''
+    + lib.optionalString rustSupport ''
+      cargoSetupPostPatchHook() { true; }
+    '';
+
+  cmakeFlags =
+    [
+      "-DENABLE_CHDIG=OFF"
+      "-DENABLE_TESTS=OFF"
+      "-DENABLE_DELTA_KERNEL_RS=0"
+      "-DCOMPILER_CACHE=disabled"
+    ]
+    ++ lib.optional (
+      stdenv.hostPlatform.isLinux && stdenv.hostPlatform.isAarch64
+    ) "-DNO_ARMV81_OR_HIGHER=1";
+
+  env = {
+    CARGO_HOME = "$PWD/../.cargo/";
+    NIX_CFLAGS_COMPILE =
+      # undefined reference to '__sync_val_compare_and_swap_16'
+      lib.optionalString stdenv.hostPlatform.isx86_64 " -mcx16"
+      +
+        # Silence ``-Wimplicit-const-int-float-conversion` error in MemoryTracker.cpp and
+        # ``-Wno-unneeded-internal-declaration` TreeOptimizer.cpp.
+        lib.optionalString stdenv.hostPlatform.isDarwin
+          " -Wno-implicit-const-int-float-conversion -Wno-unneeded-internal-declaration";
+  };
+
+  # https://github.com/ClickHouse/ClickHouse/issues/49988
+  hardeningDisable = [ "fortify" ];
+
+  postInstall = ''
+    rm -rf $out/share/clickhouse-test
```

Comment:
It is usually advisable to only use -f if required, so that the build breaks if something changes on an update.